### ISSUE TITLE
stackoverflow while comparing expressions to keywords

### ISSUE THREAD

**Issue by daslu** [sicmutils/sicmutils#291]
Hi! I ran across the following problem (discovered it while trying to render Sicmutils code in Notespace):

```bash
$ clj -Sdeps '{:deps {sicmutils/sicmutils {:mvn/version "0.15.0"}}}' 
Clojure 1.10.1
user=> (require '[sicmutils.env :as env])
nil
user=> (env/bootstrap-repl!)
WARNING: * already refers to: #'clojure.core/* in namespace: user, being replaced by: #'sicmutils.env/*
WARNING: ref already refers to: #'clojure.core/ref in namespace: user, being replaced by: #'sicmutils.env/ref
WARNING: - already refers to: #'clojure.core/- in namespace: user, being replaced by: #'sicmutils.env/-
WARNING: zero? already refers to: #'clojure.core/zero? in namespace: user, being replaced by: #'sicmutils.env/zero?
WARNING: / already refers to: #'clojure.core// in namespace: user, being replaced by: #'sicmutils.env//
WARNING: partial already refers to: #'clojure.core/partial in namespace: user, being replaced by: #'sicmutils.env/partial
WARNING: + already refers to: #'clojure.core/+ in namespace: user, being replaced by: #'sicmutils.env/+
WARNING: compare already refers to: #'clojure.core/compare in namespace: user, being replaced by: #'sicmutils.env/compare
nil
user=> (= (tan 'x) :a)
Execution error (StackOverflowError) at sicmutils.value/eval2349$fn (value.cljc:227).
null
user=> 
```

**Comment by sritchie** [comment]
Thanks for the report, @daslu! I've got a fix for this up in #292, so you should be all set shortly.

**Comment by daslu** [comment]
Wow, thanks for the quick help @sritchie !
